public/js/components/DataPropertyField.vue

```diff
@@ -0,0 +1,30 @@
+<template>
+  <label v-bind:class="{ invalid: !isValid }">
+    Data Property (supports
+    <a href="http://jmespath.org/tutorial.html" target="_blank">JMESPath</a>)
+    <input type="text" v-bind:name="name" v-model="model" v-bind:aria-invalid="!isValid" />
+    <span v-if="!isValid">This is not a valid JMESPath.</span>
```

Comment:
Extra accessibility suggestion: We could give this element an ID as well (using the same unique ID technique as suggested above) and then add something like:
```
:aria-describedby="!isValid ? `data-property-error-${tileId}` : ''"
```
to the input element. This would associate the error message with the input when the input is invalid.

Not a blocker for merging, just a nice-to-have! 😄

Reply:
Great suggestion, I'll try it. Should this `span` move outside the `label` too?

Reply:
This is an interesting scenario. For one thing, messages informing a user about invalid input should be part of the label because that causes them to be read out when a screen reader is faced with the related control. However, as @escherina suggested, we already create a relation between the validation message and the form control. I’m not quite sure if we’re over-complicating matters with this or not.

In short, I’d definitely keep the message inside the label to ensure it’s announced along with the primary label content.